test/workers/work-failer.ts

```diff
@@ -2,116 +2,171 @@ import { describe } from 'mocha';
 import MockDate from 'mockdate';
```

Comment:
It would be nice to have an end-to-end test with the `ignore-errors` flag as well to make sure that the job status is set correctly after all retries from the work failer have failed. Right now I think we have all the functionality tested since you are reusing the same code path `processWorkItemUpdate` for these updates so it's not a must to add this test, only if there's a straightforward way to add one end to end test.

Reply:
☑️